xbmc/addons/AddonInstaller.cpp

```diff
@@ -614,12 +614,15 @@ bool CAddonInstallJob::DoWork()
   if (!Install(installFrom, m_repo))
     return false;
 
-  if (!CAddonMgr::GetInstance().LoadAddon(m_addon->ID()))
+  AddonPtr addon = CAddonMgr::GetInstance().LoadAddon(m_addon->ID());
```

Comment:
Please don't misuse null to mean 'error'. That's what true/false is for. If AddonInstaller needs an instance of the new addon it can call CAddonMgr::GetAddon afterwards

Reply:
I change it, thought to hold the `CAddonMgr::GetAddon` calls smaller.